leakcanary-analyzer/src/main/java/com/squareup/leakcanary/LeakTrace.kt

```diff
@@ -0,0 +1,81 @@
+package com.squareup.leakcanary
+
+import com.squareup.leakcanary.Reachability.Status.REACHABLE
+import com.squareup.leakcanary.Reachability.Status.UNKNOWN
+import com.squareup.leakcanary.Reachability.Status.UNREACHABLE
+import java.io.Serializable
+
+/**
+ * A chain of references that constitute the shortest strong reference path from a leaking instance
+ * to the GC roots. Fixing the leak usually means breaking one of the references in that chain.
+ */
+data class LeakTrace(
+  val elements: List<LeakTraceElement>,
+  val expectedReachability: List<Reachability>
+) : Serializable {
+
+  override fun toString(): String {
+    var leakInfo = "┬\n"
+    val lastElement = elements.last()
+    val lastReachability = expectedReachability.last()
+    elements.dropLast(1)
+        .forEachIndexed { index, leakTraceElement ->
+          val currentReachability = expectedReachability[index]
+
+          leakInfo += """
+        |├─ ${leakTraceElement.className}
+        |${LINE}${getReachabilityString(currentReachability)}
+        |${LINE}${getPossibleLeakString(currentReachability, leakTraceElement, index)}
+        |
+        """.trimMargin()
+        }
+    leakInfo += """╰→ ${lastElement.className}
+      | ${getReachabilityString(lastReachability)}
+    """.trimMargin()
+
+    return leakInfo
+  }
+
+  fun toDetailedString(): String {
+    var string = ""
```

Comment:
why not string builder here?

Reply:
[`+=` is compiled to StringBuilder calls](https://discuss.kotlinlang.org/t/stringbuilder-and-kotlin/441/11)


I do need to change this part though. A better way would be using `joinToString`